mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -726,3 +801,83 @@ func testAccMongoDBAtlasClusterAWSConfigdWithLabels(projectID, name, backupEnabl
 		}
 	`, projectID, name, backupEnabled, tier, region, labelsConf)
 }
+
+func testAccMongoDBAtlasClusterConfigWithPrivateEndpointLink(awsAccessKey, awsSecretKey, projectID, providerName, region, vpcID, subnetID, securityGroupID, clusterName string) string {
+	return fmt.Sprintf(`
+		provider "aws" {
+			region     = "${lower(replace("%[5]s", "_", "-"))}"
+			access_key = "%[1]s"
+			secret_key = "%[2]s"
+		}
+
+		resource "mongodbatlas_private_endpoint" "test" {
+			project_id    = "%[3]s"
+			provider_name = "%[4]s"
+			region        = "%[5]s"
+		}
+
+		resource "aws_vpc_endpoint" "ptfe_service" {
+			vpc_id             = "%[6]s"
+			service_name       = "${mongodbatlas_private_endpoint.test.endpoint_service_name}"
+			vpc_endpoint_type  = "Interface"
+			subnet_ids         = ["%[7]s"]
+			security_group_ids = ["%[8]s"]
+		}
+
+		resource "mongodbatlas_private_endpoint_interface_link" "test" {
+			project_id            = "${mongodbatlas_private_endpoint.test.project_id}"
+			private_link_id       = "${mongodbatlas_private_endpoint.test.private_link_id}"
+			interface_endpoint_id = "${aws_vpc_endpoint.ptfe_service.id}"
+		}
+
+		resource "mongodbatlas_cluster" "test" {
+		  project_id             = "%[3]s"
+		  name                   = "%[9]s"
+		  disk_size_gb           = 5
+
+		  //Provider Settings "block"
+		  provider_name               = "AWS"
+		  provider_region_name        = "${upper(replace("%[5]s", "-", "_"))}"
+		  provider_instance_size_name = "M10"
+		  provider_backup_enabled     = true // enable cloud provider snapshots
+		  provider_disk_iops          = 100
+		  provider_encrypt_ebs_volume = false
+		  depends_on = ["mongodbatlas_private_endpoint_interface_link.test"]
+		}
+	`, awsAccessKey, awsSecretKey, projectID, providerName, region, vpcID, subnetID, securityGroupID, clusterName)
+}
+
+func testAccMongoDBAtlasClusterConfigAzureWithNetworkPeering(projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName, clusterName string) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_network_container" "test" {
+			project_id   		  = "%[1]s"
+			atlas_cidr_block  = "192.168.208.0/21"
+			provider_name		  = "%[2]s"
+			region    			  = "US_EAST_2"
```

Comment:
possible to make this dynamic as well? like what we did for `testAccMongoDBAtlasClusterConfigWithPrivateEndpointLink`?